controllers/componentsubscription_controller.go

```diff
@@ -264,19 +269,16 @@ func (r *ComponentSubscriptionReconciler) reconcile(ctx context.Context, obj *v1
 		}
 	}()
 
-	if obj.Spec.Destination != nil {
-		rreconcile.ProgressiveStatus(false, obj, meta.ProgressingReason, "transferring component to target repository: %s", obj.Spec.Destination.URL)
-
-		if err := r.OCMClient.TransferComponent(ctx, octx, obj, sourceComponentVersion, latestSourceComponentVersion.Original()); err != nil {
-			err := fmt.Errorf("failed to transfer components: %w", err)
-			status.MarkNotReady(r.EventRecorder, obj, v1alpha1.TransferFailedReason, err.Error())
+	if r.MpasEnabled {
```

Comment:
do we need if/else here? could we not just do the validation and signature calculation if mpas is enabled and then use the same reconcile logic from that point onwards.

Reply:
Yeah, I'll try to simplify this a bit. Let me see what I can cook up. :)

Reply:
Ah thanks. Turns out I overcomplicated it a bit... If it's an mpas component we already check if destination is there. So the code below didn't need change at all anyways. It's a lot nicer now.